src/operations/blas3/gemm_no_local_full_vec.hpp

```diff
@@ -0,0 +1,911 @@
+    const index_t local_item_id_row = item_id % wg_rows * vector_ofs;
+    /* work item id per column */
+    const index_t local_item_id_col =
+        (item_id / wg_rows) * (trans_b ? vector_ofs : 1);
+
+    /* Exiting from any threads outside of the m and n boundary */
+    const bool out_of_range = ((local_item_id_row + wg_row >= m) ||
+                               (local_item_id_col + wg_col >= n));
+
+    m -= local_item_id_row + wg_row;
+    n -= local_item_id_col + wg_col;
+    /*
+     * The ma and nb are used to adjust the start position of each work-item for
+     * A, B and C matrices.
+     */
+    const index_t dim_m_a_start = (local_item_id_row + wg_row);
+    const index_t dim_n_b_start = (local_item_id_col + wg_col);
+
+    /*! @brief Adjusting the start position of A, B , and C */
+    orig_A += dim_m_a_start * (trans_a ? lda : 1);
+    orig_B += dim_n_b_start * (trans_b ? 1 : ldb);
+    orig_C += dim_m_a_start + (dim_n_b_start * ldc);
+
+    /*
+     * The following lambdas: boundary_check_m, boundary_check_n, and
+     * boundary_check_c  are used to check the A, B , and C boundaries
+     * respectively.
+     */
+    const auto boundary_check_m =
+        [&](const index_t &idx) SYCL_BLAS_ALWAYS_INLINE { return idx < m; };
+    const auto boundary_check_n =
+        [&](const index_t &idx) SYCL_BLAS_ALWAYS_INLINE { return idx < n; };
+    const auto boundary_check_c =
+        [&](const index_t &dim_m_c_start, const index_t &dim_n_c_start)
+            SYCL_BLAS_ALWAYS_INLINE {
+              return (dim_m_c_start < original_m && dim_n_c_start < original_n);
+            };
+
+    // computing the next element for a and b;
+    const index_t A_ptr_index = (trans_a ? lda : 1) * wg_rows * vector_ofs;
+    const index_t B_ptr_index = (trans_b ? vector_ofs : ldb) * wg_cols;
+
+    /*
+     * computing the gemm panel
+     */
+    if ((is_internal_block == true)) {
+      compute_gemm_no_shared_pannel<false, packetize_t::packet_size>(
+          orig_A, orig_B, orig_C, a_size, b_size, c_size, a_.get_size_col(), k,
+          dim_m_a_start, dim_n_b_start, A_ptr_index, B_ptr_index,
+          boundary_check_m, boundary_check_n, boundary_check_c, out_of_range,
+          batch_stride, wg_batch_id, batch_size_, lda, ldb, ldc
+#ifdef ARM_GPU
+          ,
+          id
+#endif
+      );
+    } else {
+      compute_gemm_no_shared_pannel<true, 1>(
+          orig_A, orig_B, orig_C, a_size, b_size, c_size, a_.get_size_col(), k,
+          dim_m_a_start, dim_n_b_start, A_ptr_index, B_ptr_index,
+          boundary_check_m, boundary_check_n, boundary_check_c, out_of_range,
+          batch_stride, wg_batch_id, batch_size_, lda, ldb, ldc
+#ifdef ARM_GPU
+          ,
+          id
+#endif
+      );
+    }
+  }
+  /** @brief If beta is not zero then this function will load in values from C,
+  multiply them by the beta value and store them in the results register. If
+  beta is zero then this function does nothing. */
```

Comment:
> this function does nothing

Strictly speaking this function sets the values in the results register to zero.

Reply:
Good catch, fixed!